File: HW1/Shape.java

import java.util.Comparator;

public abstract class Shape {
	
	private String name;
	private int x, y, z; // co ordinate on screen

	public Shape (String name, int nx, int ny, int nz) {
		this.x = nx;
		this.y = ny;
		this.z = nz;
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public int getX() {
		return x;
	}

	public int getY() {
		return y;
	}

	public int getZ() {
		return z;
	}

	public void display() {
		System.out.print ("[ Location (x, y, z) = (" + x + ", " + y + ", " + z + ") ");
	}

    public String toString() {
        return "[ Location (x, Y, z) = (" + x + ", " + y + "," + z +  ")" + "]";
    }
}

File: HW2/Shape.py
class Shape(object):

  	def __init__(self, name, nx, ny, nz):
  		self._name = name
  		self._x = nx
  		self._y = ny
  		self._z = nz

	def getName(self):
		return self._name

	def getX(self):
		return self._x

	def getY(self):
		return self._y

	def getZ(self):
		return self._z

	""" Define a method to find out area of any given shape. 
		Method definition only- the derived classes will implement 
		the method
	"""
	def getArea(self):
		pass

	def display(self):
		print "[ Location (x, y, z) = (" + str(self._x) + ", " + str(self._y) + ", " + str(self._z) + ") ]",
